packages/adapters-cli/src/commands/library-commands.ts

```diff
@@ -40,11 +44,56 @@ export function libraryCommands(program: Command) {
         const filterProductIds = multiProductFilter(productIds)
         const filterChainIds = multiChainFilter(chains)
 
+        const resolvedUserAddress = userAddress ?? DEFAULT_USER_ADDRESS
+        if (!userAddress) {
+          console.log(
+            `No user address provided. Using default address ${DEFAULT_USER_ADDRESS}`,
+          )
+        }
+
         const filterProtocolTokens =
           multiProtocolTokenAddressFilter(protocolTokens)
 
+        if (!buildPoolFilter() && !filterProtocolTokens) {
```

Comment:
Only call `buildPoolFilter()` once, otherwise it's creating a bunch of pools each time.